src/openfermion/utils/_trotter_exp_to_qgates.py

```diff
@@ -0,0 +1,329 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+"""Module to perform Trotter-Suzuki decompositions to output as circuits."""
+
+from openfermion.ops import QubitOperator
+from openfermion.utils import count_qubits
+import numpy
+
+"""
+Description:
+    Functions for estimating the exponential of an operator
+    composed of Pauli matrices, by Trotterization. Outputs QASM
+    format to a python stream.
+
+    Change-of-basis gates:
+    H           : Z to X basis (and back)
+    Rx(pi/2)    : Z to Y basis .....
+    Rx(-pi/2)   : Y to Z basis .....
+"""
+
+
+def third_order_trotter_helper(op_a, op_b):
+    """Recursively trotterize a QubitOperator according
+        to the scheme: e^(A+B) = e^(7/24 * A) e^(2/3 * B) e^(3/4 * A)
+        e^(-2/3 * B) e^(-1/24 * A) e^(B)
+
+    Note:
+        See N. Hatano and M. Suzuki Lect. Notes Phys 679, 37-68 (2005)
+
+    Args:
+        op_a: (QubitOperator) first term in hamiltonian
+        op_b: (list of QubitOperator) the rest of the terms in the
+                hamiltonian as a list of QubitOperators
+    Returns:
+        list of QubitOperators giving the trotterized hamiltonian
+    """
+    # Create list to hold return value
+    ret_val = []
+
+    # e^(7/24 * A)
+    ret_val.append(7.0/24.0 * op_a)
+
+    # e^(2/3 * B), recurse if B is not a single operator
+    if len(op_b) == 1:
+        ret_val.append(2.0 / 3.0 * op_b[0])
+    else:
+        btemp = []
+        for op in op_b[1:]:
+            btemp.append(op * 2.0/3.0)
+        atemp = op_b[0] * 2.0/3.0
+        ret_val.extend(third_order_trotter_helper(atemp, btemp))
```

Comment:
If you know how to do so, you should manually remove this recursion and make it iterative.  There are some other workarounds that exist, but generically python has no tail-call optimization and using recursion for sufficiently large Hamiltonians will cause stack overflows.

Reply:
It seems difficult, but I'm working on it.

Reply:
Is it OK if I leave it for now?

Reply:
Just kidding I implemented it